synapse/rest/media/v1/media_repository.py

```diff
@@ -919,10 +919,13 @@ async def _apply_media_retention_rules(self) -> None:
             await self.delete_old_local_media(
                 before_ts=local_media_threshold_timestamp_ms,
                 keep_profiles=True,
+                delete_quarantined_media=False,
             )
 
     async def delete_old_remote_media(self, before_ts: int) -> Dict[str, int]:
-        old_media = await self.store.get_remote_media_before(before_ts)
+        old_media = await self.store.get_remote_media_ids(
+            before_ts, include_quarantined_media=False
+        )
```

Comment:
Hm, there's a bit of a discrepancy here in that `delete_old_local_media` gives you a choice of whether you want to delete quarantined media or not, whereas `delete_old_remote_media` decides for you (always false).

This arose from the fact that `delete_old_local_media` had an admin API attached to it which may want to sprout a new query parameter for whether quarantined media is deleted, whereas the remote media side had no such API.

It'd probably still be nice to align the two though.